src/mlpack/tests/callback_test.cpp

```diff
@@ -135,4 +138,34 @@ BOOST_AUTO_TEST_CASE(LRWithOptimizerCallback)
     BOOST_REQUIRE_GT(stream.str().length(), 0);
 }
 
+/**
+ * Test a Linear SVM model with EndOptimization callback.
+ */
+BOOST_AUTO_TEST_CASE(LinearSVMCallbackTest)
+{
+  const size_t numClasses = 2;
+  const double lambda = 0.0001;
+  const double delta = 1.0;
+
+  // A very simple fake dataset
+  arma::mat dataset = "2 0 0;"
+                      "0 0 0;"
+                      "0 2 1;"
+                      "1 0 2;"
+                      "0 1 0";
+
+  //  Corresponding labels
+  arma::Row<size_t> labels = "1 0 1";
+
+  CallbackTestFunction cb;
+
+  ens::L_BFGS opt;
+  LinearSVM<arma::mat> lsvm(dataset, labels, numClasses, lambda,
+      delta, false, opt, cb);
+
+  lsvm.Train(dataset, labels, numClasses, opt, cb);
```

Comment:
I think so you can remove this because train function is also calling in above constructor.